selfdrive/ui/qt/widgets/controls.h

```diff
@@ -196,6 +196,80 @@ class ParamControl : public ToggleControl {
   bool confirm = false;
   bool store_confirm = false;
 };
+ 
+class ButtonParamControl : public AbstractControl {
+  Q_OBJECT
+public:
+  ButtonParamControl(const QString &param, const QString &title, const QString &desc, const QString &icon,
+                     std::vector<QString> button_texts, std::vector<int> button_widths) : AbstractControl(title, desc, icon) {
+    select_style = (R"(
+        padding: 0;
+        border-radius: 50px;
+        font-size: 45px;
+        font-weight: 500;
+        color: #E4E4E4;
+        background-color: #33Ab4C;
+      )");
+    unselect_style = (R"(
+        padding: 0;
+        border-radius: 50px;
+        font-size: 45px;
+        font-weight: 350;
+        color: #E4E4E4;
+        background-color: #393939;
+      )");
+    key = param.toStdString();
+    for (int i = 0; i < button_texts.size(); i++) {
+      QPushButton *button = new QPushButton();
+      button->setText(button_texts[i]);
+      hlayout->addWidget(button);
+      button->setFixedSize(button_widths[i], 100);
+      button->setStyleSheet(unselect_style);
+      buttons.push_back(button);
+      QObject::connect(button, &QPushButton::clicked, [=]() {
+        params.put(key, (QString::number(i)).toStdString());
+        refresh();
+      });
+    }
+    refresh();
+  }
+
+  void read_param() {
+    auto value = params.get(key);
+    if (!value.empty()) {
+      param_value = std::stoi(value);
+    }
```

Comment:
@deanlee you think it would be a good idea to make a params class function that handles this and catches errors if stoi fails?